docs/tutorials/send-payjoin-with-pdk.md

```diff
@@ -0,0 +1,374 @@
+    pub fn try_payjoin(pj_uri: payjoin::Uri<'static, NetworkChecked>) -> Txid { }
+}
+```
+
+##### 3.2 Implement `get_sender_wallet` function inside `sender` module.
+
+Write a utility function to get the sender wallet, using `bitcoincore_rpc` to
+connect to the wallet. The `Auth` struct is used to authenticate the
+connection, and the `Client` struct is used to connect to the wallet.
+
+```rust
+// payjoin--tutorial/src/main.rs
+// mod sender {
+    fn get_sender_wallet() -> bitcoincore_rpc::Client {
+        let auth = bitcoincore_rpc::Auth::UserPass(
+            "foo".to_string(),
+            "qDDZdeQ5vw9XXFeVnXT4PZ--tGN2xNjjR4nrtyszZx0=".to_string(),
+        );
+        bitcoincore_rpc::Client::new("http://localhost:18443/wallet/sender", auth).unwrap()
+    }
+    // ..
+// }
+```
+
+##### 3.3 Implement `https_agent` function inside `sender` module.
+
+The `https_agent` function is used in order to make https requests.
+The sender will use this function to send the payjoin request to the
+receiver.
+
+Here, we are using `danger_accept_invalid_certs` as we run this
+locally. In production this should be initiated with valid
+certificates only.
+
+[Why to use HTTPS in production](https://www.cloudflare.com/learning/ssl/why-is-http-not-secure/)
+
+For Payjoin, it is important to use HTTPS to ensure the privacy of the
+transaction and to prevent potential lose of funds.  Without
+authenticated communications, an attacker could spoof the payjoin
+endpoint and steal funds. Without encrypted communications, anyone on
+the network can see the contents of the payjoin and foil privacy.
+
+```rust
+// payjoin--tutorial/src/main.rs
+// mod sender {
+    // ..
+    fn https_agent() -> reqwest::blocking::Client {
+        let https = reqwest::blocking::Client::builder()
+            .danger_accept_invalid_certs(true)
+            .build()
+            .expect("failed to build https client");
+        https
+    }
+    // ..
+// }
+```
+
+In the next few steps, we will add the core implementation for the
+sender inside the `try_payjoin` function.
+
+##### 3.4 Extract the parameters from the payjoin URI
+
+Payjoin requires messages to be sent between sender and receiver. The
+receiver hosts an HTTP endpoint to receive these messages. To make a
+request, we need the receiver payjoin Uri. 
+
+The Uri contains the amount the receiver is requesting and the address
+they wish to receive to. Moreover, the Uri also includes the endpoint
+we will make the request to, and that is extracted in the
+`RequestContext::from_psbt_and_uri`.
+
+```rust
+// mod sender {
+    // ..
+    pub fn try_payjoin(pj_uri: payjoin::Uri<'static, NetworkChecked>) -> Txid {
+        let sender_wallet = get_sender_wallet();
+        let amount_to_send = pj_uri.amount.unwrap();
+        let receiver_address = pj_uri.address.clone();
+    }
+// }
+```
+
+##### 3.4 Construct an "Original PSBT" paying `.amount` to `.address`
+
+as specified in the URI.
```

Comment:
I know I suggested that a space be added after headings, but perhaps this line is this actually meant to be part of the heading?

It seems you're using automatic newlines, and I don't believe markdown will render headings on multiple lines. When I write markdown I just have my editor wordwrap and write long lines.

Reply:
yea it was a mistaken line-fixing. 

removed totally the .amount and .address from header as they are mentioned in the body just next to it.